bugfixtemplates/getting-substring-before-suffix/GettingSubstringBeforeSuffix.java

```diff
@@ -0,0 +1,135 @@
+package bugfixtemplates;
+
+class GettingSubstringBeforeSuffix {
+
+    Object mutex, mutex1;
+
+    public void m1(String className) {
+        int i = className.lastIndexOf(SUFFIX);
+        className.substring(0, i);
+    }
+
+    public void m2(String className) {
+        int i = className.lastIndexOf(SUFFIX);
+        String subclassPrefix = i > 0 ? className.substring(0, i) : className;
+    }
+
+    public void m3(String className) {
+        int i = className.indexOf(SUFFIX);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 18 does not find the substring and returns -1, the call to substring() on line 19 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string className.